test/BytesSignature.t.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "../src/libraries/ByteSignature.sol";
+import {UserOp} from "./helpers/UserOp.sol";
+
+contract ByteSignatureTest is UserOp {
+    function testExtractSingleSignature() public {
+        // dummy signature
+        bytes memory signature =
+            hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff01";
+        bytes[] memory extracted = ByteSignature.extractSignatures(signature, 1);
+        assertEq(extracted[0], signature, "The extracted signature does not match the original signature.");
+    }
+
+    function testExtractMultipleSignatures() public {
+        // create 2 dummy signatures
+        bytes memory signature1 =
+            hex"ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff01";
+        bytes memory signature2 =
+            hex"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaabbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb1c";
+        bytes memory fullSignature = abi.encodePacked(signature1, signature2);
+
+        bytes[] memory extracted = ByteSignature.extractSignatures(fullSignature, 2);
+        assertEq(extracted[0], signature1, "The first signature does not match the original signature.");
+        assertEq(extracted[1], signature2, "The second signature does not match the original signature.");
+    }
```

Comment:
can we add another test with 3 just in case? It should work

Reply:
i've added a fuzz test so we test any amount of signatures